.github/workflows/build-android.yml

```diff
@@ -0,0 +1,55 @@
+name: Test Android Pull Requests
+
+on:
+  pull_request:
+    branches: [ main ]
+    paths:
+      - 'android/**'
+
+  push:
+    branches: [ main ]
+    paths:
+      - 'android/**'
+
+  # This allows us to manually run this job
+  workflow_dispatch:
+
+jobs:
+
+  kotlin-code-checks:
+    name: Code Tests
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: gradle
+
+      - name: Grant execute permission for gradlew (Mobile)
```

Comment:
FYI, you can fix the `chmod` issue locally and upload it in the PR. File permissions are stored in git